code/__HELPERS/lists.dm

```diff
@@ -53,10 +53,16 @@
 
 //Checks for specific types in a list
 /proc/is_type_in_list(atom/A, list/L)
+	if (!L.len)
+		return 0
+	if (!isnum(L[L[1]]))
+		generate_type_list_cache(L)

```

Comment:
there are only 37 places this proc is called so you can probably just grab teh lists being called and audit for all instances of that list